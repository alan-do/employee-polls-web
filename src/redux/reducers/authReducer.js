import { GRANTED_AUTHORIZATION, DENIED_AUTHORIZATION } from "../actions/authAction";

export default function authReducer(state = null, action) {
    switch (action.type) {
        case GRANTED_AUTHORIZATION:
            return action.authenticatedUser;
        case DENIED_AUTHORIZATION:
            return null;
        default:
            return state;
    }
}
