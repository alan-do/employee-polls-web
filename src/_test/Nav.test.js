import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import { BrowserRouter as Router } from "react-router-dom";
import React from "react";
import Nav from "../components/navigation/Navigation";
import authReducer from "../redux/reducers/authReducer";

const createMockStore = (initialState) => {
  return configureStore({
    reducer: {
      authenticatedUser: authReducer,
    },
    preloadedState: initialState,
  });
};

describe("Nav", () => {
    it("should render the component", () => {
        const mockStore = createMockStore({
          authenticatedUser: { id: "sarahedo", password: "" }
        });

        render(
            <Provider store={mockStore}>
                <Router>
                    <Nav/>
                </Router>
            </Provider>
        );
        expect(screen.getByRole("navigation")).toBeInTheDocument();
        expect(screen.getByRole("navigation")).toMatchSnapshot();
    });

    it("should display username of logged in user", () => {
        const mockStore = createMockStore({
          authenticatedUser: { id: "sarahedo", password: "" }
        });

        render(
            <Provider store={mockStore}>
                <Router>
                    <Nav/>
                </Router>
            </Provider>
        );

        const userInfoElement = screen.getByTestId("user-information");
        expect(userInfoElement.textContent).toBe("sarahedo");
    });
});
