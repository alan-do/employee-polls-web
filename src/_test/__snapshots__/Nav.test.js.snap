// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render the component 1`] = `
<nav
  class="nav-container"
>
  <div
    class="nav-items"
  >
    <a
      class="nav-item selected"
      href="/"
    >
      Dashboard
    </a>
    <a
      class="nav-item "
      href="/leaderboard"
    >
      Leaderboard
    </a>
    <a
      class="nav-item "
      href="/add"
    >
      New Poll
    </a>
  </div>
  <div
    class="user-info"
  >
    <img
      alt="User Avatar"
      class="user-avatar"
    />
    <span
      class="user-name"
      data-testid="user-information"
    >
      sarahedo
    </span>
    <button
      class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
    >
      Logout
    </button>
  </div>
</nav>
`;
